gb-cpu/src/error.rs

```diff
@@ -1,5 +1,6 @@
 #[derive(Debug)]
 pub enum Error {
-    InvalidAbsoluteAddress(u16),
     InvalidPC(u16),
+    SetError(usize, u8),
```

Comment:
What's `SetError` ? What it represent ? *so much question*

Reply:
I've a ref list for my editor, but I'm reading the PR on github.
If a name don't say anything to me alone : it's a bad name

> The name of a variable, function, or class, should answer all the big questions. It should tell you why it exists, what it does, and how it is used. If a name requires a comment, then the name does not reveal its intent.
> #CleanCode

Reply:
I think having Error in the name of a variant of an enum called `Error` is redundant.  

As FirelightFlagboy point out, `SetError` might not be helpful to understand what the error is about. Perhaps something like `ForbiddenWrite` or `IllegalSet` would be more explicit ?

Reply:
My trait Bus has a function Set and this function Set can have an error =  SetError. Nothing crazy here.

Reply:
@guilhemSmith Great Point, that's way better than SetError